src/subgraph/dex/client.ts

```diff
@@ -0,0 +1,29 @@
+import { Maybe, TokenInfo } from "../../types";
+import { getLogger } from "../../utilities";
+
+import * as actions from "./actions";
+import { createApolloClient } from "../helpers";
+
+const logger = getLogger("subgraph:dexClient");
+
+export interface UniswapSubgraphClient {
```

Comment:
Does this need to be named as uniswap specific? Or could this be a generic interface like **TokenSubgraphClient**

Reply:
Specifically for the story this is for it only needs to add Uniswap. We can consider a more abstract base interface / class in the future when adding more DEX protocols but they will likely have at least mildly different schemas and will need to each be named uniquely